enumerate moabs in directory

jmartin-sul opened the pull request:
the `MoabStorageDirectory` class offers a couple of methods that allow storage root traversal and listing.  each takes a directory that contains a druid tree (such as a storage trunk in a storage root).

`.find_moab_paths` will allow the caller to pass in a block, which will then get the `druid`, `path`, and `MatchData` object as parameters for each druid under the specified directory.  `.list_moab_druids` provides a convenient wrapper around this method, and just returns a list of all druids in the given directory.

closes #67

tingulfsen commented on lib/benchmarking/moab_enumeration_testing.rb:
```diff
@@ -0,0 +1,110 @@
+  end
+
+  def logger
+    @logger ||= Logger.new('log/benchmark.log')
```
Any chance we could use a slightly more descriptive name here, i.e. a separate log file for this benchmarker?

jmartin-sul commented on config/initializers/moab_config.rb:
```diff
@@ -0,0 +1,8 @@
+# TODO: is this the best place for these require statements?  seems like app without it being somewhere.
```
ugh, thought i'd fixed that typo, but apparently not.  anyway, if you're fine with that placement, i'll just remove it entirely.  i thought maybe there might be a more appropriate or canonically rails-y place to put it, like application.rb or something, but i couldn't find any clear guidance on that when searching our apps or the web in general.

jmartin-sul commented:
@tingulfsen thanks for the feedback, will get rid of the todo with the typo, will differentiate the benchmark log files, and will post a short summary of the benchmarking from the other approaches i tested.

jmartin-sul commented:
@tingulfsen:  i think i've addressed your concerns, and i pushed a rebased version of the branch with the requested changes.  there's some data about the performance testing i did in the 3rd commit (https://github.com/sul-dlss/preservation_core_catalog/pull/115/commits/1a4299b2f4e82db0edf98f636fd22bb17e41cf5b).  for more gory detail, including the code that was actually benchmarked, and a bit more commentary on the performance, see this branch (which was cleaned up and reworked into the branch being used for this PR):  https://github.com/sul-dlss/preservation_core_catalog/compare/67-enumerate-moabs_benchmarking

jmartin-sul commented:
note that in the above linked benchmarking branch, [one of the approaches](https://github.com/sul-dlss/preservation_core_catalog/compare/67-enumerate-moabs_benchmarking#diff-535d55a7301d74539efa687da1812252R47) does this weird thing where it tries to construct the whole space of possible druid trees, but it only actually does it for part of the name space.  this was just to keep it from running for longer than was tractable to analyze, since it seemed like it might not complete in a timely manner after initial testing.
